File: p1.c
#include <stdio.h>
#include <stdlib.h>

typedef struct {
    int* arr;
    int capacity;
    int topIndex;
} myStack;

myStack* createStack(int n) {
    myStack* stack = (myStack*)malloc(sizeof(myStack));
    stack->capacity = n;
    stack->topIndex = -1;
    stack->arr = (int*)malloc(n * sizeof(int));
    return stack;
}

int isEmpty(myStack* stack) {
    return stack->topIndex == -1;
}

int isFull(myStack* stack) {
    return stack->topIndex == stack->capacity - 1;
}

void push(myStack* stack, int x) {
    if (!isFull(stack)) {
        stack->arr[++stack->topIndex] = x;
    }
}

void pop(myStack* stack) {
    if (!isEmpty(stack)) {
        stack->topIndex--;
    }
}

int peek(myStack* stack) {
    if (isEmpty(stack)) return -1;
    return stack->arr[stack->topIndex];
}

File: p1.py
class myStack:
    def __init__(self, n):
        self.size = n
        self.arr = [0] * n
        self.top = -1

    def isEmpty(self):
        return self.top == -1

    def isFull(self):
        return self.top == self.size - 1

    def push(self, x):
        if not self.isFull():
            self.top += 1
            self.arr[self.top] = x

    def pop(self):
        if self.isEmpty():
            return -1
        val = self.arr[self.top]
        self.top -= 1
        return val

    def peek(self):
        if self.isEmpty():
            return -1
        return self.arr[self.top]
